src/platform/Darwin/PlatformManagerImpl.cpp

```diff
@@ -53,6 +53,27 @@ CHIP_ERROR PlatformManagerImpl::_InitChipStack()
     return err;
 }
 
+CHIP_ERROR PlatformManagerImpl::_StartEventLoopTask()
+{
+    dispatch_resume(mWorkQueue);
+
+    return CHIP_NO_ERROR;
+};
+
+CHIP_ERROR PlatformManagerImpl::_StopEventLoopTask()
+{
+    dispatch_sync(mWorkQueue, ^{
+        dispatch_suspend(mWorkQueue);
+    });
+
+    return CHIP_NO_ERROR;
+};
+
+void PlatformManagerImpl::_RunEventLoop()
+{
+    _StartEventLoopTask();
```

Comment:
This is not sufficient - it has to block on this call and not return till the event loop has been signalled to terminate.

Reply:
I have added a call to `CFRunLoopMain()` which will block.

Reply:
That's definitely better - you'd need to add a call to `CFRunLoopStop` in `_StopEventLoopTask` to properly cause its termination however.